File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/ScanSample/src/com/hp/workpath/sample/scansample/fragments/MarginsPreferenceFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.scansample.fragments;

import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.preference.PreferenceDialogFragmentCompat;

import com.hp.workpath.sample.scansample.R;

import static com.hp.workpath.sample.scansample.MainActivity.TAG;

public class MarginsPreferenceFragment extends PreferenceDialogFragmentCompat {

    EditText mLeftMarginEditText;
    EditText mTopMarginEditText;
    EditText mRightMarginEditText;
    EditText mBottomMarginEditText;

    public static MarginsPreferenceFragment newInstance(String key) {
        final MarginsPreferenceFragment fragment = new MarginsPreferenceFragment();
        final Bundle bundle = new Bundle();
        bundle.putString(ARG_KEY, key);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    protected void onBindDialogView(@NonNull View view) {
        super.onBindDialogView(view);
        mLeftMarginEditText = view.findViewById(R.id.leftEditText);
        mTopMarginEditText = view.findViewById(R.id.topEditText);
        mRightMarginEditText = view.findViewById(R.id.rightEditText);
        mBottomMarginEditText = view.findViewById(R.id.bottomEditText);
        mLeftMarginEditText.setText(Float.toString(getMarginsPreference().getLeft()));
        mTopMarginEditText.setText(Float.toString(getMarginsPreference().getTop()));
        mRightMarginEditText.setText(Float.toString(getMarginsPreference().getRight()));
        mBottomMarginEditText.setText(Float.toString(getMarginsPreference().getBottom()));
    }

    // get the NumberPickerPreference instance
    private MarginsPreference getMarginsPreference() {
        return (MarginsPreference) getPreference();
    }

    @Override
    public void onDialogClosed(boolean positiveResult) {
        if (positiveResult) {
            try {
                getMarginsPreference().setMargins(getFloatValue(mLeftMarginEditText),
                        getFloatValue(mTopMarginEditText),
                        getFloatValue(mRightMarginEditText),
                        getFloatValue(mBottomMarginEditText));
                getMarginsPreference().applyMargins();
            } catch (Throwable t) {
                Log.e(TAG, "getMarginsPreference " + t.getMessage());
            }
        }
    }

    private float getFloatValue(EditText editText) {
        float value = 0.0f;
        try {
            if (editText != null && !TextUtils.isEmpty(editText.getText().toString())) {
                value = Float.parseFloat(editText.getText().toString());
            }
        } catch (Throwable t) {}
        return value;
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/ScanSample/src/com/hp/workpath/sample/scansample/fragments/MarginsPreferenceFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.scansample.fragments

import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.preference.PreferenceDialogFragmentCompat
import com.hp.workpath.sample.scansample.MainActivity
import com.hp.workpath.sample.scansample.R

class MarginsPreferenceFragment : PreferenceDialogFragmentCompat() {

    lateinit var mLeftMarginEditText: EditText
    lateinit var mTopMarginEditText: EditText
    lateinit var mRightMarginEditText: EditText
    lateinit var mBottomMarginEditText: EditText

    override fun onBindDialogView(view: View?) {
        super.onBindDialogView(view)
        view?.run {
            mLeftMarginEditText = findViewById(R.id.leftEditText)
            mTopMarginEditText = findViewById(R.id.topEditText)
            mRightMarginEditText = findViewById(R.id.rightEditText)
            mBottomMarginEditText = findViewById(R.id.bottomEditText)

            mLeftMarginEditText.setText(marginsPreference.left.toString())
            mTopMarginEditText.setText(marginsPreference.top.toString())
            mRightMarginEditText.setText(marginsPreference.right.toString())
            mBottomMarginEditText.setText(marginsPreference.bottom.toString())
        }
    }

    // get the NumberPickerPreference instance
    private val marginsPreference: MarginsPreference
        get() = preference as MarginsPreference

    override fun onDialogClosed(positiveResult: Boolean) {
        if (positiveResult) {
            try {
                marginsPreference.setMargins(getFloatValue(mLeftMarginEditText),
                        getFloatValue(mTopMarginEditText),
                        getFloatValue(mRightMarginEditText),
                        getFloatValue(mBottomMarginEditText))
                marginsPreference.applyMargins()
            } catch (t: Throwable) {
                Log.e(MainActivity.TAG, "getMarginsPreference ${t.message}")
            }
        }
    }

    private fun getFloatValue(editText: EditText?): Float {
        var value = 0.0f
        try {
            if (editText != null && !TextUtils.isEmpty(editText.text.toString())) {
                value = editText.text.toString().toFloat()
            }
        } catch (e: Exception) {
        }
        return value
    }

    companion object {
        fun newInstance(key: String?): MarginsPreferenceFragment {
            val fragment = MarginsPreferenceFragment()
            val bundle = Bundle()
            bundle.putString(ARG_KEY, key)
            fragment.arguments = bundle
            return fragment
        }
    }
}